cluster.go

```diff
@@ -51,7 +51,7 @@ type member struct {
 }
 
 // newCluster returns a new marathon cluster
-func newCluster(client *http.Client, marathonURL string) (*cluster, error) {
+func newCluster(client *http.Client, marathonURL string, DCOSPath string) (*cluster, error) {
```

Comment:
The problem with `registerSSESubscription` (and the SSE part in general) is that it's fairly detached from the regular cluster management logic. This really needs an overhaul outside of the scope of this PR.

Nevertheless, I believe we can fix things in a minimalistic fashion by moving the cluster member selection into `buildAPIRequest` and adjust the call sites accordingly. This allows us to change `newCluster` to respect a custom DCOS path in the Marathon URL while still defaulting to `/marathon` when needed. It's unfortunate that we have to extend `newCluster`'s signature; however, I realized this function needs to change anyway along a bigger refurbishment, so I guess it's okay for now.

Hope you didn't mind that I pushed a commit of mine onto your branch. Please have a look and let me know if it makes sense to you.